lib/galaxy/managers/groups.py

```diff
@@ -47,33 +35,33 @@ def index(self, trans: ProvidesAppContext):
         for group in get_not_deleted_groups(trans.sa_session):
             item = group.to_dict(value_mapper={"id": DecodedDatabaseIdField.encode})
             encoded_id = DecodedDatabaseIdField.encode(group.id)
-            item["url"] = url_for("group", id=encoded_id)
+            item["url"] = self._url_for(trans, "index", group_id=encoded_id)
```

Comment:
These URLs don't look correct. It should be returning something like: `/api/groups/{encoded_group_id}` but this one is returning ``/api/groups/{encoded_group_id}/roles`. You need to play with the `name` of the FastAPI route to make it generate the route you need.